pkg/core/decorate.go

```diff
@@ -18,26 +18,32 @@ package core
 // some non-objects (such as PodTemplates) define annotations but are not objects.
 type Annotated interface {
 	GetAnnotations() map[string]string
-	SetAnnotations(annotations map[string]string)
+	SetAnnotations(map[string]string)
 }
 
 // SetAnnotation sets the annotation on the passed annotated object to value.
-func SetAnnotation(obj Annotated, annotation, value string) {
-	as := obj.GetAnnotations()
-	if as == nil {
-		as = make(map[string]string)
+// Returns true if the object was modified.
+func SetAnnotation(obj Annotated, annotation, value string) bool {
```

Comment:
Can you add unit tests for these functions?

Reply:
Done.

Reply:
Thanks! looks good